laba1.c

```diff
@@ -0,0 +1,242 @@
+               num,
+               BLUE,
+               CLOSE);
+        break;
+    case ERROR_UNEXPECT_TOKEN:
+        printf("Error at column %d: unexpected token\n", num);
+        break;
+    case ERROR_EXPECT_COMMA:
+        printf("Error at column %d: expected %s','%s\n", num, BLUE, CLOSE);
+        break;
+    }
+}
+
+void to_lower(char* str, int num)
+{
+    for (int i = 0; i < num; i++) {
+        str[i] = tolower(str[i]);
+    }
+}
+
+double coordinat_x(char* arr, int* num)
+{
+    char exp[20];
+    int i = 0;
+
+    while (!isdigit(arr[*num]) && arr[*num] != '-') {
+        if (arr[*num] == '(' || arr[*num] == ' ') {
+            *num += 1;
+        } else {
+            if (arr[*num] == ')') {
+                show_error(ERROR_BRACKET, *num, &arr[*num]);
+                exit(EXIT_FAILURE);
+            } else {
+                show_error(ERROR_NUMBER, *num, NULL);
+                exit(EXIT_FAILURE);
+            }
+        }
+    }
+
+    while (isdigit(arr[*num]) || arr[*num] == '-' || arr[*num] == '.') {
+        exp[i] = arr[*num];
+        i++;
+        *num += 1;
+    }
+    if (arr[*num] != ' ') {
+        show_error(ERROR_UNEXPECT_TOKEN, *num, NULL);
+        exit(EXIT_FAILURE);
+    }
+    char* trash;
+    return strtod(exp, &trash);
+}
+
+double coordinat_y(char* arr, int* num)
+{
+    char exp[20];
+    int i = 0;
+
+    while (!isdigit(arr[*num]) && arr[*num] != '-') {
+        if (arr[*num] == ' ') {
+            *num += 1;
+        } else {
+            show_error(ERROR_NUMBER, *num, NULL);
+            exit(EXIT_FAILURE);
+        }
+    }
+
+    while (isdigit(arr[*num]) || arr[*num] == '-' || arr[*num] == '.') {
+        exp[i] = arr[*num];
+        i++;
+        *num += 1;
+    }
+
+    while (arr[*num] != ',') {
+        if (arr[*num] == ' ') {
+            *num += 1;
+        } else {
+            show_error(ERROR_EXPECT_COMMA, *num, NULL);
+            exit(EXIT_FAILURE);
+        }
+    }
+    char* trash;
+    return strtod(exp, &trash);
+}
+
+double radius_search(char* arr, int* num)
+{
+    char exp[20];
+    int i = 0;
+
+    while (!isdigit(arr[*num])) {
+        if (arr[*num] == ' ' || arr[*num] == ',') {
+            *num += 1;
+        } else {
+            show_error(ERROR_NUMBER, *num, NULL);
+            exit(EXIT_FAILURE);
+        }
+    }
+
+    while (isdigit(arr[*num]) || arr[*num] == '.') {
+        exp[i] = arr[*num];
+        i++;
+        *num += 1;
+    }
+
+    while (arr[*num] != ')') {
+        if (arr[*num] == ' ') {
+            *num += 1;
+        } else {
+            if (arr[*num] == '(') {
+                show_error(ERROR_BRACKET, *num, &arr[*num]);
+                exit(EXIT_FAILURE);
+            } else {
+                show_error(ERROR_UNEXPECT_TOKEN, *num, NULL);
+                exit(EXIT_FAILURE);
+            }
+        }
+    }
+    char* trash;
+    return strtod(exp, &trash);
+}
+
+void is_end_empty(char* arr, int* num)

```

Comment:
Может что-то вроде такого названия придумать?
```suggestion
void is_tail_empty(char* arr, int* num)
```

Раз имеется в виду проверка чего-нибудь за закрывающей скобкой.